plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -242,6 +247,8 @@ data class ClassTypeEmbedding(val className: ScopedKotlinName) : TypeEmbedding {
     private var _predicate: Predicate? = null
     val fields: Map<SimpleKotlinName, FieldEmbedding>
         get() = _fields ?: throw IllegalStateException("Fields of $className have not been initialised yet.")
+    val allFields: List<FieldEmbedding>
```

Comment:
Perhaps deduplicating here already would make sense?  I think I had some code that works for all fields already, could that be reused?  (But maybe I'm misremembering and that's for all supertypes or something.)

Reply:
I can reuse some code instead of `allFields` but the deduplication cannot be put just there because doing this:
```
        val getSuperPropertyFunctions = classSuperTypes.flatMap {
            it.allFields
            ...
```
will still have duplicates due to the fact that fields are taken from all super types